operations/agent_monitor.go

```diff
@@ -384,44 +372,45 @@ func waitUntilComplete(ctx context.Context, proc jasper.Process, maxDelay time.D
 
 // runAgent starts the agent with the necessary args and waits for it to
 // terminate.
-func runAgent(ctx context.Context, m *monitor, retry *retryArgs) error {
-	proc, err := createAgentProcess(ctx, m, retry)
+func (m *monitor) runAgent(ctx context.Context, retry util.RetryArgs) error {
+	proc, err := m.createAgentProcess(ctx, retry)
 	if err != nil {
 		return errors.Wrapf(err, "failed to create agent process")
 	}
 
-	exitCode, err := waitUntilComplete(ctx, proc, defaultRetryArgs().maxDelay)
+	exitCode, err := waitUntilComplete(ctx, proc, defaultMaxRequestDelay)
 
 	return errors.Wrapf(err, "agent exited with code %d", exitCode)
 }
 
 // runMonitor runs the monitor loop. It fetches the agent, starts it, and
 // repeats when the agent terminates.
-func runMonitor(ctx context.Context, m *monitor) {
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-			// The evergreen agent runs using a separate binary from the monitor
-			// to allow the agent to be updated.
-			if err := fetchClient(ctx, m, defaultRetryArgs()); err != nil {
-				grip.Error(message.WrapError(err, message.Fields{
-					"message":     "could not fetch client",
-					"client_url":  m.clientURL,
-					"client_path": m.clientPath,
-				}))
-				continue
-			}
+func (m *monitor) run(ctx context.Context) {
+	for ; ctx.Err() == nil; time.Sleep(defaultMaxRequestDelay) {
```

Comment:
Would it make sense to use `util.Retry` here instead? I believe that handles contexts, and it provides exponential backoff with jitter, which will help be nice to the app server in case of error.

Reply:
Add `util.Retry()`.